### ISSUE TITLE
Editing plugins in build.bnd requires eclipse restart.

### ISSUE THREAD

**Issue by tomq42** [bndtools/bndtools#1440]
Steps to reproduce.

New workspace
Create bnd workspace using osgi/workspace archetype
New Bnd Osgi Project, "OSGi enRoute" template. Call it foo.bar
Let it all compile.

Paste the following in to the cnf/build.bnd file:

```
-plugin: \
	aQute.bnd.deployer.repository.FixedIndexedRepo;name=Amdatu Dependencies;locations=http://repository.amdatu.org/dependencies/index.xml.gz,\
	aQute.bnd.deployer.repository.FixedIndexedRepo;name=Amdatu Release;locations=http://repository.amdatu.org/release/index.xml.gz
```

Code now all goes red. Repositories view only has "Workspace xxx". 
Exit eclipse and restart.
REpository view is now fully populated again. 
Project, clean all projects. 
Compile errors all go away.

Deleting the changes from the build.bnd file also requires you to restart eclipse.

Maybe, what I'm pasting in is invalid (in which case more graceful response would be better), maybe it';s supposed to work this way (in which case, surprise), maybe this is a bug.

**Comment by gamerson** [comment]
I'll take a look.

**Comment by gamerson** [comment]
@tomq42 can you try to reproduce with the very latest developmental build from here? 

https://bndtools.ci.cloudbees.com/job/bndtools.master/lastSuccessfulBuild/artifact/build/generated/p2/

**Comment by gamerson** [comment]
I'll close this for now, but if you are able to reproduce with the latest bndtools, please be sure to reopen and I'll take a look.

**Comment by tomq42** [comment]
I've just had the same issue again with 3.3.0.DEV-20160805-165325-gb2a3af3.
My build.bnd file contained:

   -plugin: \
	aQute.bnd.deployer.repository.FixedIndexedRepo;name=Amdatu Dependencies;locations=http://repository.amdatu.org/dependencies/index.xml.gz,\
	aQute.bnd.deployer.repository.FixedIndexedRepo;name=Amdatu Release;locations=http://repository.amdatu.org/release/index.xml.gz

and I deleted the first line ("Amdatu Dependencies".

**Comment by tomq42** [comment]
In my .metadata/.log file I have:

!ENTRY bndtools.builder 1 0 2016-08-11 08:54:35.115
!MESSAGE Unable to get workspace
!STACK 0
java.lang.IllegalStateException: Central is not initialised
	at bndtools.central.Central.getWorkspace(Central.java:198)
	at org.bndtools.builder.CnfWatcher.resourceChanged(CnfWatcher.java:40)
	at org.eclipse.core.internal.events.NotificationManager$1.run(NotificationManager.java:299)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.NotificationManager.notify(NotificationManager.java:289)
	at org.eclipse.core.internal.events.NotificationManager.broadcastChanges(NotificationManager.java:152)
	at org.eclipse.core.internal.resources.Workspace.broadcastPostChange(Workspace.java:373)
	at org.eclipse.core.internal.resources.Workspace.endOperation(Workspace.java:1470)
	at org.eclipse.core.internal.resources.Marker.delete(Marker.java:72)
	at org.eclipse.debug.core.model.Breakpoint.delete(Breakpoint.java:137)
	at org.eclipse.wst.jsdt.debug.internal.core.JavaScriptPreferencesManager.stop(JavaScriptPreferencesManager.java:113)
	at org.eclipse.wst.jsdt.debug.internal.core.JavaScriptDebugPlugin.stop(JavaScriptDebugPlugin.java:187)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$4.run(BundleContextImpl.java:827)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$4.run(BundleContextImpl.java:1)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.stop(BundleContextImpl.java:820)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.stopWorker0(EquinoxBundle.java:955)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.stopWorker(EquinoxBundle.java:323)
	at org.eclipse.osgi.container.Module.doStop(Module.java:626)
	at org.eclipse.osgi.container.Module.stop(Module.java:488)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.decStartLevel(ModuleContainer.java:1623)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1542)
	at org.eclipse.osgi.container.SystemModule.stopWorker(SystemModule.java:248)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$SystemBundle$EquinoxSystemModule.stopWorker(EquinoxBundle.java:144)
	at org.eclipse.osgi.container.Module.doStop(Module.java:626)
	at org.eclipse.osgi.container.Module.stop(Module.java:488)
	at org.eclipse.osgi.container.SystemModule.stop(SystemModule.java:186)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$SystemBundle$EquinoxSystemModule$1.run(EquinoxBundle.java:159)
	at java.lang.Thread.run(Unknown Source)

!ENTRY bndtools.builder 1 0 2016-08-11 08:54:35.118
!MESSAGE Unable to get workspace
!STACK 0
java.lang.IllegalStateException: Central is not initialised
	at bndtools.central.Central.getWorkspace(Central.java:198)
	at org.bndtools.builder.CnfWatcher.resourceChanged(CnfWatcher.java:40)
	at org.eclipse.core.internal.events.NotificationManager$1.run(NotificationManager.java:299)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.NotificationManager.notify(NotificationManager.java:289)
	at org.eclipse.core.internal.events.NotificationManager.broadcastChanges(NotificationManager.java:152)
	at org.eclipse.core.internal.resources.Workspace.broadcastPostChange(Workspace.java:373)
	at org.eclipse.core.internal.resources.Workspace.endOperation(Workspace.java:1470)
	at org.eclipse.core.internal.resources.Marker.delete(Marker.java:72)
	at org.eclipse.wst.jsdt.debug.internal.core.JavaScriptPreferencesManager.stop(JavaScriptPreferencesManager.java:124)
	at org.eclipse.wst.jsdt.debug.internal.core.JavaScriptDebugPlugin.stop(JavaScriptDebugPlugin.java:187)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$4.run(BundleContextImpl.java:827)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$4.run(BundleContextImpl.java:1)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.stop(BundleContextImpl.java:820)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.stopWorker0(EquinoxBundle.java:955)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.stopWorker(EquinoxBundle.java:323)
	at org.eclipse.osgi.container.Module.doStop(Module.java:626)
	at org.eclipse.osgi.container.Module.stop(Module.java:488)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.decStartLevel(ModuleContainer.java:1623)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1542)
	at org.eclipse.osgi.container.SystemModule.stopWorker(SystemModule.java:248)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$SystemBundle$EquinoxSystemModule.stopWorker(EquinoxBundle.java:144)
	at org.eclipse.osgi.container.Module.doStop(Module.java:626)
	at org.eclipse.osgi.container.Module.stop(Module.java:488)
	at org.eclipse.osgi.container.SystemModule.stop(SystemModule.java:186)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$SystemBundle$EquinoxSystemModule$1.run(EquinoxBundle.java:159)
	at java.lang.Thread.run(Unknown Source)

**Comment by bjhargrave** [comment]
> My build.bnd file contained:
> 
>    -plugin: \
>     aQute.bnd.deployer.repository.FixedIndexedRepo;name=Amdatu Dependencies;locations=http://repository.amdatu.org/dependencies/index.xml.gz,\
>     aQute.bnd.deployer.repository.FixedIndexedRepo;name=Amdatu Release;locations=http://repository.amdatu.org/release/index.xml.gz

Your repo names have spaces in them and you do not quote the names. This could be an issue: `name="Amdatu Release"` or `name=AmdatuRelease`.

**Comment by tomq42** [comment]
I *think* that I created the lines originally using the plugin editor, rather than typing it in manually.
Either way though, it all works fine with those lines in the file. Start eclipse up, and I get the two repositories in the repository explorer. It only goes wrong when I delete one of them.

So if that *is* the problem, then there's some issues somewhere validating or handling that.

(Yes, I've just tried it again. If I edit the build.bnd file and use the "workspace tab" rather than the "source" tab, and I add a new Plugin, give the name a string with spaces in, lets me do it, the name in the bnd file is not quoted).
